### PR TITLE
add native stake command: ioctl stake2 create

### PR DESCRIPTION
**Pull request by RuneAkko**
#2017
@dustinxie

### CODE REVIEW

File: ioctl/cmd/action/stake2.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"github.com/spf13/cobra"
+
+	"github.com/iotexproject/iotex-core/ioctl/config"
+)
+
+// Multi-language support
+var (
+	stake2CmdUses = map[config.Language]string{
+		config.English: "stake2",
+		config.Chinese: "stake2",
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2.go:19]: add usage like current v1

File: ioctl/cmd/action/stake2.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	}
+)
+
+var stake2AutoRestake bool
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2.go:23]: autoRestakeV2?

File: ioctl/cmd/action/stake2.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	"github.com/iotexproject/iotex-core/ioctl/config"
+)
+
+// Multi-language support
+var (
+	stake2CmdUses = map[config.Language]string{
+		config.English: "stake2",
+		config.Chinese: "stake2",
+	}
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2.go:20]: do we need stakeCmdShorts, flagEndpointUsages, flagInsecureUsages like current code?

File: ioctl/cmd/action/stake2create.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+func init() {
+	registerWriteCommand(stake2CreateCmd)
+	stake2CreateCmd.Flags().BoolVar(&stake2AutoRestake, "auto-restake", false,
+		config.TranslateInLang(stake2FalgAutoRestakeUsages, config.UILanguage))
+}
+
+func stake2Create(args []string) error {
+	var amount = args[0]
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2create.go:55]: this should also need StringToRau?

File: ioctl/cmd/action/stake2create.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+	var candidateName = args[1]
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2create.go:56]: we also need to validate the name, will let u know the detail method soon

Comment (dustinxie) [reply ioctl/cmd/action/stake2create.go:56]: refer to staking/validations.go

func IsValidCandidateName(s string) bool

File: ioctl/cmd/action/stake2create.go (hunk 4 above)
Comment (RuneAkko) [reply ioctl/cmd/action/stake2create.go:55]: action.NewCreateStake function accepts the "amount" as string?

Comment (RuneAkko) [reply ioctl/cmd/action/stake2create.go:55]: > this should also need StringToRau?

action.NewCreateStake function accepts the "amount" as string?

File: ioctl/cmd/action/stake2create.go (hunk 5 above)
Comment (RuneAkko) [reply ioctl/cmd/action/stake2create.go:56]: > refer to staking/validations.go
> 
> func IsValidCandidateName(s string) bool

ok, I will change ioctl/validator/validator.go:func ValidateCandidateName() to this func.

File: ioctl/cmd/action/stake2create.go (hunk 4 above)
Comment (dustinxie) [reply ioctl/cmd/action/stake2create.go:57]: yes, but as far as i remember unit should be Rau?

For example, staking 100 token, args[0] from input is "100", but amount to call NewCreateStake () should be "100000000000000000000"

can you double-check that's the case?

File: ioctl/validator/validator.go [hunk 6]
```diff
@@ -71,9 +71,13 @@ func ValidateStakeDuration(stakeDuration *big.Int) error {
 
 // ValidateCandidateName validates candidate name for native staking
 func ValidateCandidateName(candidateName string) error {
-	if len(candidateName) > 12 {
-		return ErrLongCandidateName
+	if len(s) == 0 || len(s) > 12 {
```
Comment (dustinxie) [inline ioctl/validator/validator.go:74]: current ValidateCandidateName() is also used for V1, should not change it

i think we'll need a new ValidateCandidateNameV2() for staking V2

File: ioctl/validator/validator.go [hunk 7]
```diff
@@ -24,7 +24,7 @@ var (
 	// ErrInvalidStakeDuration indicates error for invalid stake duration
 	ErrInvalidStakeDuration = errors.New("stake duration must be within 0 and 1050 and in multiples of 7")
 	// ErrInvalidCandidateName indicates error for invalid candidate name
-	ErrLongCandidateName = errors.New("invalid length of candidate name that is more than 12 ")
+	ErrCandidateName = errors.New("the candidate name string is not valid")
```
Comment (dustinxie) [inline ioctl/validator/validator.go:27]: i think define a new Err for V2, see comment below

File: ioctl/cmd/action/stake2create.go (hunk 4 above)
Comment (RuneAkko) [reply ioctl/cmd/action/stake2create.go:57]: ok，but it seems that codes in line 49  in action.NewCreateStake function will be duplicated.

File: ioctl/cmd/action/stake2create.go [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+	gasPriceRau, err := gasPriceInRau()
+	if err != nil {
+		return output.NewError(0, "failed to get gas price", err)
+	}
+	nonce, err := nonce(sender)
+	if err != nil {
+		return output.NewError(0, "failed to get nonce ", err)
+	}
+
+	s2c, err := action.NewCreateStake(nonce, candidateName, amountStringInRau, duration, stake2AutoRestake, data, gasLimit, gasPriceRau)
```
Comment (dustinxie) [inline ioctl/cmd/action/stake2create.go:100]: no need for amountStringInRau := strconv.FormatUint(amountInRau.Uint64(), 10) in above
just use amountInRau.String()

**Pull request merged**